scale/recipe/models.py

```diff
@@ -215,12 +216,14 @@ def get_recipe_for_job(self, job_id):
         :rtype: :class:`recipe.models.Recipe`
         """
 
-        recipe_job_qry = RecipeJob.objects.select_related('recipe__recipe_type', 'recipe__recipe_type_rev')
-        try:
-            recipe_job = recipe_job_qry.get(job_id=job_id, is_original=True)
-        except RecipeJob.DoesNotExist:
+        job_recipe = []
+        for recipe_job in RecipeJob.objects.filter(job_id=job_id, is_original=True):
+            # This should match at most one RecipeJob object
+            job_recipe.append(recipe_job)
+
+        if not job_recipe:
             return None
-        return recipe_job.recipe
+        return job_recipe[0]
```

Comment:
This loses the related recipe_type and recipe_type_rev models, which we want. Also the original code is closer to what we want. All that is needed here is to change `return recipe_job.recipe` to `return recipe_job`